### ISSUE TITLE
Optional Parameters with TypeScript and Flow 

### ISSUE THREAD

**Issue by johnnycoyleST** [mobxjs/mobx#1569]
I have a question about using mobx 4's async flow in conjunction with TypeScript (strict mode). What is the best way to handle optional parameters being passed to the generator function. Here's an example of what I'm talking about:

```
    init = flow(function * (foo: string, bar?: number) {
	//do some asychronous action
    });
```

The above would lead to the compilation error: 

`"TS2554: Expected 1 arguments, but got 2" `

I was able to get around this by wrapping flow in a HoF that handles the optional params. Is there another solution to this, or has that functionality not yet been integrated for flow?

**Comment by kuitos** [comment]
Can't reproduce the compilation error with mobx 4.2.0, ts 2.8.1
```ts
const init = flow(function *(name: string, age?: number) {
	yield console.log(name);
	yield console.log(age);
});
```
can u provide a jsfiddle link or any other similar demos?

**Comment by kanoshin** [comment]
@kuitos just try to call function that you created with 2 parameters and you would get an error:
<img width="478" alt="screen shot 2018-06-07 at 3 14 53 pm" src="https://user-images.githubusercontent.com/11817360/41129133-94295c86-6a65-11e8-8f32-01025059811e.png">

**Comment by Youzhigang** [comment]
the same to me.

**Comment by dassennato** [comment]
Same here!

**Comment by MatthieuCoelho** [comment]
Same here

**Comment by mweststrate** [comment]
Potentially fixed by #1816
